contracts/src/fp/OPSuccinctFaultDisputeGame.sol

```diff
@@ -407,51 +477,50 @@ contract OPSuccinctFaultDisputeGame is Clone, ISemver {
             status = GameStatus.CHALLENGER_WINS;
             resolvedAt = Timestamp.wrap(uint64(block.timestamp));
 
-            // Record the challenger's reward
-            credit[claimData.counteredBy] += address(this).balance;
+            // Add challenger's reward to the entry point contract.
+            (bool success,) = ENTRY_POINT.call{value: address(this).balance}(
+                abi.encodeWithSelector(
+                    OPSuccinctEntryPoint.addCredit.selector, claimData.counteredBy, address(this).balance
+                )
+            );
+            if (!success) revert BondTransferFailed();
 
             emit Resolved(status);
         } else if (claimData.status == ProposalStatus.UnchallengedAndValidProofProvided) {
             claimData.status = ProposalStatus.Resolved;
             status = GameStatus.DEFENDER_WINS;
             resolvedAt = Timestamp.wrap(uint64(block.timestamp));
 
-            // Record the proposer's reward
-            credit[gameCreator()] += address(this).balance;
+            // Return the proposer's bond to the entry point contract.
```

Comment:
I would be explicit when saying it's returned to the entry point contract about who it's being returned to.